python/pyspark/sql/dataframe.py

```diff
@@ -588,7 +588,7 @@ def withWatermark(self, eventTime, delayThreshold):
 
         Parameters
         ----------
-        eventTime : str or :class:`Column`
+        eventTime : str
```

Comment:
The scalar eventTime is also `str`
https://github.com/apache/spark/blob/61d038f26e1beeeb39638fdf9703d2e86d058342/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L753
and python implements is also validated as string
https://github.com/apache/spark/blob/ff1fc5ed4b685b4f5f83d5f600b24f089dd4522e/python/pyspark/sql/dataframe.py#L608-L609

but wrong doc on the param and typehints